### ISSUE TITLE
Investigation on ram

### ISSUE THREAD

**Issue by jimmykane** [janrueth/SiriServerCore#14]
After some beers, just as i came home the python had stalled on the Public server! Hehe. 
Ok no error, just stalled. The server went out of ram. 
Reason:  Screenshot after 3 exceptions (handled crashes) This screen shot is new! not a log check (edit)
http://dl.dropbox.com/u/2374229/Screenshot%20at%202012-03-11%2004%3A06%3A04.png
I did a log check and found out that after some usual (not connection related) crashes eg key erros etc (unhandled) that the older python process was not killed (quit). Thus taking more and more ram after every crash. 

I might be mistaken and this might be totally connection related but i am new to python and I am not sure of Core Concepts. 

Also sometimes after some load or some plugins usage the ctrl+C wont exit right and stall at last step. 
This might be related to the above ram consumption. I think so at least. Because it's the only reason I can think.

**Comment by jimmykane** [comment]
Edited

**Comment by janrueth** [comment]
Hm... I don't know how python does memory management, I have a look at that

**Comment by janrueth** [comment]
I've added a couple of unsettings of variables... it could be that there was a cyclic reference there which caused the garbage collector to not remove a connection and its associated variables... also increasing ram size does not directly go along with the memory usage of the python process, if python frees memory it is not directly free for the system to use, python  (sometimes?) reserves it for further allocations and thus the memory footprint is not decreasing.

I won't close the issue as I cannot say that it is resolved

**Comment by janrueth** [comment]
Do you have any new measurements regarding this issue?

**Comment by jimmykane** [comment]
Yes. Everytime the server gets an unhandled exception then it stays on memory like an instance. 
Eg if it crashes 10 times and the bin mem size = 10 mb then total python memory consumed by the system ~=100mb 
atm only crashes at the lock acquire
